### PR TITLE
(PE-30041) Remove harmful term from pcp-broker config

### PR DESCRIPTION
**Pull request by donoghuc**
This commit changes instances of `whitelist` to `allowlist`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ The major updates are:
   inventory updates, which were introduced in the [PCP v2
   specifications](https://github.com/puppetlabs/pcp-specifications/blob/master/pcp/versions/2.0/inventory.md#inventory-messages)
 * [PCP-681](https://tickets.puppetlabs.com/browse/PCP-681) Add the
-  `controller-uris`  and `controller-whitelist` configuration options. These
+  `controller-uris`  and `controller-allowlist` configuration options. These
```
Comment (steveax) [inline CHANGELOG.md:134]: Seems like we should not retroactively change that note, but rather add a new change to the Changelog detailing the name change?

**Review by steveax — COMMENTED** [review]
Just one nit on the changelog. It's not 100% clear to me if we need to allow for the legacy setting name though. Will an upgrade take care of remapping the configuration keys?

**Review by barriserloth — APPROVED** [review]
I'm +1 on this, but should we add a note about the change to the changelog like Steve was suggesting?

**Comment by steveax** [comment]
@barriserloth, that can be a separate prepare for release commit.

**Review by steveax — APPROVED** [review]
👍

**Pull request merged**